engine/src/gui/text_area.cpp

```diff
@@ -105,8 +106,8 @@ TextArea::TextArea(float x, float y, float wid, float hei, int scrollbar) {
     vertical_left_of_text = 0.02;
     text_spacing = (font_size_float / 100) + (horizontal_spacer * 2);
 
-    //The parent TextAreaItem. This is the only link where parent == NULL. It is not displayed and handled only internally
-    ItemList = new TextAreaItem("", "", NULL);
+    //The parent TextAreaItem. This is the only link where parent == nullptr. It is not displayed and handled only internally
+    item_tree_top = MakeShared<TextAreaItem>("", "", nullptr);
```

Comment:
Is this the real meat of the PR?

Reply:
It's certainly part of it. Along with all the lines where I replaced raw pointers with `WeakPtr` or `SharedPtr`. And the call(s) to `SharedFromThis`. And the `deque` of `SharedPtr` for child items, in place of the array of raw pointers.